commands/set_default_builder.go

```diff
@@ -21,6 +21,26 @@ func SetDefaultBuilder(logger *logging.Logger) *cobra.Command {
 				return nil
 			}
 
+			imageName := args[0]
+
+			logger.Verbose("Verifying locally...")
+			info, err := inspector.InspectBuilder(imageName, true)
```

Comment:
Does this do any validation on the builder other than existence? I need to look at what scenario a nil info object is returned...

Reply:
Yes, it's basically the same mechanism used in `pack inspect-builder` where it retrieves the build info via labels/metadata. 

The info would be nil if the image does not exist. It's not considered an error. That's how the `pack inspect-builder` command denotes "Not present".

I'll add a few snapshots of the output in various scenarios.

Reply:
I think though that we're still missing the cases where a user sets an existing but terrible image as their builder. For instance, someone recently set `java` as their builder and pack gave no feedback that it wouldn't work.
We should add an additional check that looks for something that only exists on a valid builder (like the "io.buildpacks.builder.metadata" label, possibly even validating the JSON in that label's value, though I'm not sure how far we should go).

Reply:
That's already in there. :) See output example 2. I used 'java' explicitly thinking about that use case.

Reply:
Ah I see. I think checking for stack ID is sufficient for now.

I did notice some weird output in the error message: `stackID`. Could you change it to `stack ID`? And also have it style the first builder with `style.Symbol` rather than single quotes?
I don't think that was your doing, but it's still in the domain of this story, while we're in that area.

Reply:
Requested changes made.

Reply:
This is minor too: May want to adjust the verbiage to be consistent between local and remote:
```
Verifying local image...
Verifying remote image...
```

OR

```
Verifying locally...
Verifying remotely...
```

I somewhat prefer the first example because it seems clearer, but that's just me

Reply:
I had gone with the latter yesterday (already changed) but I can update it with the former. I do like that better.

Do you know if the rebase will be merged back in? I went ahead and merged the broken master yesterday due to code conflicts before I noticed it was broken. It seems now like master was reverted. Should I revert as well or wait for rebase to go into master?

Reply:
Sounds good.

Yes, assume the rebase refactor will be there. I'm about to fix the issue (Windows build fails), and I'll remerge back. You can wait until then to avoid rebase hell.